src/lib/core/CHIPConfig.h

```diff
@@ -86,6 +86,19 @@
 #define CHIP_CONFIG_ERROR_TYPE int32_t
 #endif // CHIP_CONFIG_ERROR_TYPE
 
+/**
+ *  @def CHIP_CONFIG_CHIP_IS_SINGLETON
+ *
+ *  @brief
+ *    This defines whether chip is singleton. If chip is singleton, all
+ *    pointers to chip components like SecureSessionMgr, ExchangeManager, etc
+ *    can be omitted, which saves memory on small platforms. For some situation
+ *    where there can be multiple chip instances, define this to 0.
+ */
+#ifndef CHIP_CONFIG_CHIP_IS_SINGLETON
+#define CHIP_CONFIG_CHIP_IS_SINGLETON 1
```

Comment:
how do we test functiohnality when this is 0 or 1? Do we have separate builds validating that both versions work correctly?